### PR TITLE
Fixes gold slime pylon permanent crabulization

### LINKED ISSUE #11431: Gold slime pylon effect does not always reverse
### Severity

Round Breaking

### Short Description

After touching a gold slimic pylon, the effect of turning into a random pet did not go away after leaving the area. This left me stuck as a crab for the (relatively short) remainder of the round.

### Reproduction

1: Touch a gold slimic pylon as normal and turn into a pet
2: Move far enough away that you leave the effect area of the pylon
3: small chance it does not revert and you are stuck as a simple mob forever

### Round ID

35125

### Test Merges

#10535, #11366, #11415

### BYOND Username

Mqiib

### Additional information

A golem tested it in front of me and it worked fine for them. The effect also does not get removed after the pylon is broken.

### PR DESCRIPTION
**Pull request by Mqiib**
Fixes #11431 
Someone forgot that crabs aren't considered a /mob/living/simple_animal/**pet** and so the reversal effect never occurred for them.

# Changelog

:cl:  

bugfix: Turning into a crab via a gold slime pylon is no longer permanent

/:cl:

### CODE REVIEW

**Comment by cuackles** [comment]
Return to crab

**Comment by ChesterTheCheesy** [comment]
Carcinisation

**Comment by UselessTheremin** [comment]
christ your commits

**Review by UselessTheremin — APPROVED** [review]
code looks good to me

**Comment by JamieD1** [comment]
Are there any other mobs you want to discredit? Might be best just making a list entry instead..

**Comment by Mqiib** [comment]
> Are there any other mobs you want to discredit? Might be best just making a list entry instead..

No, it's just crabs that are the problem here.

**Comment by SomeguyManperson** [comment]
should probably move the list of choices into the effect so it can be checked when the effect is added and removed in case someone wants to add more stuff

**Comment by Mqiib** [comment]
> should probably move the list of choices into the effect so it can be checked when the effect is added and removed in case someone wants to add more stuff

What like save all the possible animals as a list and have it check that list for the on_mob_leave? I'm not super sure how to do that without shitcoding it. I COULD just slap in a comment telling anyone that changes it to be sure to adjust the on_mob_leave list as well, which would be the lazier solution.

**Comment by SomeguyManperson** [comment]
as in have list at line like 499 then have both enter and leave check that list so it's the same list

**Review by alexkar598 — DISMISSED** [review]
I realize this is technically not part of your PR but please change line 505 to be var/mob/living/simple_animel/chosen_pet to prevent potential runtimes.

File: code/modules/research/xenobiology/crossbreeding/_structures.dm [hunk 1]
```diff
@@ -514,7 +515,7 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
 		human_mob.mind.transfer_to(chosen_pet)
 
 /obj/structure/slime_crystal/gold/on_mob_leave(mob/living/affected_mob)
-	if(!istype(affected_mob,/mob/living/simple_animal/pet) && !istype(affected_mob,/mob/living/simple_animal/crab)) //crabs are not pets
+	if(!(is_type_in_list(affected_mob, gold_pet_options))
```
Comment (SomeguyManperson) [inline code/modules/research/xenobiology/crossbreeding/_structures.dm]: ( debt

Comment (Mqiib) [reply code/modules/research/xenobiology/crossbreeding/_structures.dm]: 👍

File: code/modules/research/xenobiology/crossbreeding/_structures.dm [hunk 2]
```diff
@@ -514,7 +515,7 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
 		human_mob.mind.transfer_to(chosen_pet)
 
 /obj/structure/slime_crystal/gold/on_mob_leave(mob/living/affected_mob)
-	if(!istype(affected_mob,/mob/living/simple_animal/pet))
+	if(!is_type_in_list(affected_mob, gold_pet_options)
```
Comment (SomeguyManperson) [inline code/modules/research/xenobiology/crossbreeding/_structures.dm]: kind of need this
```suggestion
	if(!is_type_in_list(affected_mob, gold_pet_options))
```

Comment (Mqiib) [reply code/modules/research/xenobiology/crossbreeding/_structures.dm]: shhSHSHhshsHSHShshhshhhhhhhhhh

**Review by SomeguyManperson — APPROVED** [review]

**Pull request merged**